src/extensions/default/AutoUpdate/unittests.js

```diff
@@ -0,0 +1,491 @@
+/*
+ * Copyright (c) 2018 - present Adobe Systems Incorporated. All rights reserved.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a
+ * copy of this software and associated documentation files (the "Software"),
+ * to deal in the Software without restriction, including without limitation
+ * the rights to use, copy, modify, merge, publish, distribute, sublicense,
+ * and/or sell copies of the Software, and to permit persons to whom the
+ * Software is furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+ * FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+ * DEALINGS IN THE SOFTWARE.
+ *
+ */
+
+/*global describe, runs, beforeEach, it, expect, waitsFor, beforeFirst, afterLast */
+/*unittests: AutoUpdate*/
+
+define(function (require, exports, module) {
+    'use strict';
+
+    var SpecRunnerUtils = brackets.getModule("spec/SpecRunnerUtils"),
+        FileSystem = brackets.getModule("filesystem/FileSystem"),
+        Strings = brackets.getModule("strings"),
+        FileUtils = brackets.getModule("file/FileUtils");
+
+    var AutoUpdate,
+        testWindow,
+        appInitDone,
+        updateDir,
+        installerPath,
+        errorString,
+        fileCheckCompleted,
+        downloadCompleted,
+        updateBarPresent,
+        updateBarDismissed,
+
+        UPDATE_BAR = "#update-bar",
+        UPDATE_LATER_BTN = "#update-btn-later",
+        UPDATE_RESTART_BTN = '#update-btn-restart',
+        CLOSE_ICON = "#close-icon",
+
+        _modulePath = FileUtils.getNativeModuleDirectoryPath(module),
+        _unittestFilesFolderName = "unittest-files",
+        _unittestFilesFolder = [_modulePath, _unittestFilesFolderName].join("/"),
+
+        validInstallerFilePaths = {
+            "WIN": [_unittestFilesFolder, "Brackets.test.msi"].join("/"),
+            "OSX": [_unittestFilesFolder, "Brackets.test.dmg"].join("/")
+        },
+        invalidInstallerFilePaths = {
```

Comment:
@niteskum Instead of having these test images in the repo, can we think of hosting these in an S3 bucket. That will keep the footprint of the repo low.

Reply:
Test Image size is of around 800 kbs , so I think it should be fine to have in repo.